canopen/pdo/base.py

```diff
@@ -23,9 +23,9 @@ class PdoBase(Mapping):
     """
 
     def __init__(self, node):
-        self.network = None
-        self.map = None  # instance of PdoMaps
-        self.node = node
+        self.network: Optional["Network"] = None
```

Comment:
One can avoid circular import by adding conditional imports:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from canopen.network import Network
```

And I do recommend using
```python
from __future__ import annotation

# This allows
network: Optional[Network] = None

# Instead of
network: Optional["Network"] = None
```

Reply:
I am aware of those things, I used the string-notation because it allows to type hint without actually importing the classes, at least in VSCode and PyCharm it autocompletes with string-hints.
Of course this should be consistent through the project.
What is the minimal python version supported? 3.7? Then I would prefer the TYPE_CHECKING if-clause to prevent circular imports. Another solution is local imports, but that is ugly too imho.

Reply:
With PR #428 the minimum Python version will be 3.8

I believe its not advisable to rely on typing parsing without importing it, even if some IDEs have extra logic that searches the project. E.g. a github action that runs mypy is not unrealistic.

Reply:
I changed pdo/base.py to use correct type hints now and  Iused the if-clause as workaround for cyclic imports.

Reply:
After reading up on Postponed Evaluation of Annotations (PEP-563 and PEP-649), I agree that both the `if TYPE_CHECKING` and the `__future__.annotations` import are the best way to approach this since we are probably targeting Python 3.8 for the next release.  Therefore I opened #451 to start collecting these changes.  The relevant part here should be factored out and moved to that PR instead.